CCLocation/LibraryStateHandling/Reducers/LocationReducers/CurrentGeofencesMonitoringReducer.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  CurrentGeofencesMonitoringReducer.swift
+//  CCLocation
+//
+//  Created by Mobile Developer on 20/08/2019.
+//  Copyright © 2019 Crowd Connected. All rights reserved.
+//
+
+import ReSwift
+
+private struct CurrentGeofencesMonitorinReducerConstants {
+    static let userDefaultsCurrentGeofencesMonitoringKey = "currentGeofencesMonitoringKey"
+}
+
+private typealias C = CurrentGeofencesMonitorinReducerConstants
+
+func currentGeofencesMonitoringReducer (action: Action, state: CurrentGeofencesMonitoringState?) -> CurrentGeofencesMonitoringState {
+    var state = state ?? CurrentGeofencesMonitoringState(monitoringGeofences: [])
+   
+    switch action {
+        
+    case let newGeofencesAction as EnableGeofencesMonitoringAction:
+        state.monitoringGeofences = newGeofencesAction.geofences ?? []
+        
+    case _ as DisableGeofencesMonitoringAction:
+        state.monitoringGeofences = []
+        
+    default:
+        break
+    }
+    
+    return state
+}
+
+private func getCurrentStateFromUserDefaults () -> CurrentGeofencesMonitoringState? {
+    let userDefaults = UserDefaults.standard
+    let value = userDefaults.string(forKey: C.userDefaultsCurrentGeofencesMonitoringKey)
+    
+    if value != nil {
+        return CurrentGeofencesMonitoringState(monitoringGeofences: [])
+    } else {
+        return nil
+    }
+}
+
+private func saveCurrentStateToUserDefaults(currentGEOState: CurrentGeofencesMonitoringState) {
+    // Monitored circular regions and beacon regions are not saved into UserDefaults
```

Comment:
Do we want to save this?

Reply:
I observed that beacon monitored regions are not saved as well, though the method exists. The fact is that the OS is saving the monitored regions and they persists until we ask for them to be removed. With this behaviour, I don't think it is necessary to save any of them in UserDefaults. Is it better to remove the save to UserDefaults methods for this 2 variables (containing regions)?

Reply:
I guess that was the original thinking. You could argue though that if we get to a maximum of regions in conjunction with what an app developer could do, it would be worth to check at each start to "top up" the regions, which might have become available. If we want to be consistent at this point, I would remove the methods, as not doing anything, as you suggested.